src/daemonManager.js

```diff
@@ -145,19 +145,26 @@ const generateDaemon = (swarmId, daemonConfig) => {
             }
         });
 
-        await new Promise((resolve, reject) => {
-            setTimeout(() => {
-                getDaemonProcess().kill();
-                reject(new Error(`Daemon-${daemonConfig.listener_port} failed to start in ${harnessConfigs.daemonStartTimeout}ms.`))
-            }, harnessConfigs.daemonStartTimeout);
-
+        let startupTimeout = new Promise((_, reject) => {
```

Comment:
Should you use const instead of let since you never reassign the variable?

Reply:
Yepp I should, incoming patch